### ISSUE TITLE
duplicate folder created when bookmark is created and added directly into new folder

### ISSUE THREAD

**Issue by LaurenWags** [brave/browser-android-tabs#278]
**Did you search for similar issues before submitting this one?**
Yes

**Description:**
Pyramid0 - MacOS
Pyramid1 - Nexus 5
Pyramid2 - Galaxy S4

When a bookmark is created and added into a new folder on Pyramid2, a duplicate (empty) folder is created on Pyramids 0&1

**Device Details:**
   - Install Type(ARM, x86): ARM
   - Device(Phone, Tablet, Phablet): see above
   - Android Version: Nexus5 is on 5.1, Galaxy S4 is on 5.0.1
    
**Brave Version:**
1.0.23(beta) 

**Steps to reproduce:**
1. MacOS, Nexus5, Galaxy S4 all have clean profiles and no bookmarks.
2. Create Sync group on Mac.
3. Join Sync group on Nexus 5.
4. Join Sync group on Galaxy S4.
5. On Nexus 5 create a bookmark as such: navigate to a site. Tap on the 3dots. Tap on the star. Tap 'Edit' link at the bottom of the screen. Tap on 'Mobile Bookmarks' in the middle of the screen. Tap on 'New Folder'. Enter New Folder title. Tap on checkmark in upper right corner. (new folder is now displayed where 'Mobile Bookmarks' was). Nav back to page.
6. This single folder and the bookmark it contains are synced to laptop and Galaxy S4.
7. On Galaxy S4 create a bookmark as such: navigate to a site. Tap on the 3dots. Tap on the star. Tap 'Edit' link at the bottom of the screen. Tap on 'Mobile Bookmarks' in the middle of the screen. Tap on 'New Folder'. Enter New Folder title. Tap on checkmark in upper right corner. (new folder is now displayed where 'Mobile Bookmarks' was). Nav back to page.

**Actual Behavior**
A second (empty) folder with the same name as entered in Step 7 is created on Nexus 5 (Pyramid 1) and laptop (Pyramid 0).

**Expected Behavior**
Duplicate folder should not be created.

**Is this an issue with Beta build?**
Yes

**Is this an issue in the currently released version?**
No

**Can this issue be consistently reproduced?**
Yes

**Extra QA steps:**
    1. 
    2. 
    3.

**Website problems only:**
  - did you check with Brave Shields down?
  - did you check in Chrome for same behavior?

**Screenshot if needed:**

**Any related issues:**

**Comment by SergeyZhukovsky** [comment]
Android bookmark's manager allows to have several folders with same names. If we want to disable that, it is a more complex bug that is not related to sync.
Anyway I was able to reproduce the empty folder problem, but I have it only on production, I tried a staging server and it works great there, I have two folders with the same names and each bookmark is under a folder. I suggest to wait until we have fixes related to folder hierarchy merged from sync staging to production and check that issue after that.
Let's create a new ticket not related to sync if we want deny creation of two folders with same names.
